File: tests/sample.go
// sample.go
package main

import "fmt"

type User struct {
	ID   int
	Name string
}

func add(a, b int) int {
	return a + b
}

func main() {
	users := []User{{ID: 1, Name: "Ada"}, {ID: 2, Name: "Linus"}}
	fmt.Println(users[0])
	fmt.Println(add(2, 3))
}

File: tests/sample.rs
// sample.rs
#[derive(Debug)]
struct User {
    id: i32,
    name: String,
}

fn add(a: i32, b: i32) -> i32 {
    a + b
}

fn main() {
    let users = vec![
        User { id: 1, name: "Ada".into() },
        User { id: 2, name: "Linus".into() },
    ];
    println!("{:?}", users[0]);
    println!("{}", add(2, 3));
}
